kernel/arch/i386/arch_syscalls.c

```diff
@@ -602,6 +602,7 @@ void handle_syscall(regs_t *r)
     */
    r->eflags |= EFLAGS_IF;
 
+   get_curr_task()->user_state_regs = r;
```

Comment:
Even if we end up actually needing a new regs pointer like `user_state_regs`, that would be set in `save_current_task_state()` conditionally when we're in syscall context, but I'm not convinced we need that. See my other comments below.